scripts/Diffraction/isis_powder/pearl_routines/pearl_calibration_algs.py

```diff
@@ -3,92 +3,49 @@
+    :param rebin_2_params: Parameters for the second rebin step (as a string in the usual format)
+    :param cross_correlate_params: Parameters for CrossCorrelate (as a dictionary PropertyName: PropertyValue)
+    :param get_det_offset_params: Parameters for GetDetectorOffsets (as a dictionary PropertyName: PropertyValue)
+    """
+    input_ws_list = common.load_current_normalised_ws_list(run_number_string=calibration_runs, instrument=instrument,
                                                            input_batching=INPUT_BATCHING.Summed)
-    input_ws = input_ws_list[0]
-    run_details = self._get_run_details(calibration_runs)
-
-    if run_details.instrument_version == "new" or run_details.instrument_version == "new2":
-        input_ws = mantid.Rebin(InputWorkspace=input_ws, Params="100,-0.0006,19950")
-
-    d_spacing_cal = mantid.ConvertUnits(InputWorkspace=input_ws, Target="dSpacing")
-    d_spacing_cal = mantid.Rebin(InputWorkspace=d_spacing_cal, Params="1.8,0.002,2.1")
-
-    if run_details.instrument_version == "new2":
-        cross_cor_ws = mantid.CrossCorrelate(InputWorkspace=d_spacing_cal, ReferenceSpectra=20,
-                                             WorkspaceIndexMin=9, WorkspaceIndexMax=1063, XMin=1.8, XMax=2.1)
-
-    elif run_details.instrument_version == "new":
-        cross_cor_ws = mantid.CrossCorrelate(InputWorkspace=d_spacing_cal, ReferenceSpectra=20,
-                                             WorkspaceIndexMin=9, WorkspaceIndexMax=943, XMin=1.8, XMax=2.1)
-    else:
-        cross_cor_ws = mantid.CrossCorrelate(InputWorkspace=d_spacing_cal, ReferenceSpectra=500,
-                                             WorkspaceIndexMin=1, WorkspaceIndexMax=1440, XMin=1.8, XMax=2.1)
-    if self._old_api_uses_full_paths:  # Workaround for old API setting full paths
-        grouping_file_path = grouping_file_name
-        offset_file_path = offset_file_name
-    else:
-        offset_file_path = os.path.join(self.calibration_dir, offset_file_name)
-        grouping_file_path = os.path.join(self.calibration_dir, grouping_file_name)
-
-    # Ceo Cell refined to 5.4102(3) so 220 is 1.912795
-    offset_output_path = mantid.GetDetectorOffsets(InputWorkspace=cross_cor_ws, Step=0.002, DReference=1.912795,
-                                                   XMin=-200, XMax=200, GroupingFileName=offset_file_path)
-    del offset_output_path  # This isn't used so delete it to keep linters happy
-    aligned_ws = mantid.AlignDetectors(InputWorkspace=input_ws, CalibrationFile=offset_file_path)
-    cal_grouped_ws = mantid.DiffractionFocussing(InputWorkspace=aligned_ws, GroupingFileName=grouping_file_path)
-
-    common.remove_intermediate_workspace(d_spacing_cal)
-    common.remove_intermediate_workspace(cross_cor_ws)
-    common.remove_intermediate_workspace(aligned_ws)
-    common.remove_intermediate_workspace(cal_grouped_ws)
 
+    input_ws = input_ws_list[0]
```

Comment:
I struggle to create a situation where this would happen - if loading runs fails then Load throws an error before we get here, and if the user requests a set of runs of length 0 then we get an exception elsewhere (the commit i've just pushed makes this error more informative).